File: ruby/difference-of-squares/difference_of_squares.rb
# Squares: not the ones with same sides
class Squares
  def initialize(number)
    @range = 1.upto(number)
  end

  def square_of_sum
    @range.sum.abs2
  end

  def sum_of_squares
    @range.map(&:abs2).sum
  end

  def difference
    square_of_sum - sum_of_squares
  end
end

File: typescript/difference-of-squares/difference-of-squares.ts
/**
 * Difference between squares not rectangles
 */
export default class Squares {
  constructor(private readonly _end: number) {}

  get squareOfSum(): number {
    return Math.pow((this._end * (this._end + 1 ) / 2), 2)
  }

  get sumOfSquares(): number {
    return this._end * (this._end + 1) * (2 * this._end + 1) / 6
  }

  get difference(): number {
    return this.squareOfSum - this.sumOfSquares
  }
}